lib/benchee/benchmark/measure/memory.ex

```diff
@@ -11,16 +11,21 @@ defmodule Benchee.Benchmark.Measure.Memory do
     ref = make_ref()
     Process.flag(:trap_exit, true)
     start_runner(fun, ref)
-    await_results(nil, ref)
+
+    return_value =
+      try do
+        fun.()
+      catch
+        _, _ -> nil
+      end
```

Comment:
why do we catch here, we normally don't catch for faulty functions as we then assume the benchmarking code to be faulty?

Reply:
The "parent" process is essentially a supervisor because we're trapping exits (with `Process.flag(:trap_exit, true)`), so we've been ensuring that benchee never crashes entirely because of an exception in a benchmark function. Since we're now running the benchmark in the parent process and not in a linked child, it would be possible to crash benchee if we don't catch errors here.

Reply:
I think crashing benchee is fine if the function itself crashes - that's also what we do in normal run times isn't it? I mean it'd sort of be nice not to but I'm not willing to invest the code :D 

So my benchee  should never crash is more like benchee itself should never crash - user provided code may crash and handling that imo isn't our responsibility they should fix it (even if they catch it themselves)

Reply:
Oh, ok. Cool. I'll let it crash, then!